eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -375,7 +375,17 @@ object Graph {
      * @param capacity    channel capacity
      * @param balance_opt (optional) available balance that can be sent through this edge
      */
-    case class GraphEdge(desc: ChannelDesc, update: ChannelUpdate, capacity: Satoshi, balance_opt: Option[MilliSatoshi])
+    case class GraphEdge(desc: ChannelDesc, update: ChannelUpdate, capacity: Satoshi, balance_opt: Option[MilliSatoshi]) {
+
+      def maxHtlcAmount(reservedCapacity: MilliSatoshi): MilliSatoshi = Seq(
+        balance_opt.map(balance => balance - reservedCapacity),
+        update.htlcMaximumMsat,
+        Some(capacity.toMilliSatoshi - reservedCapacity)
+      ).flatten.min.max(0 msat)
+
```

Comment:
I'll dive more into it, but I think we should be ok, since #1446 made sure that `capacity` is set to `htlcMaximumMsat` instead of `0` for mobile. I'll investigate and will report back.

Reply:
I don't think it interferes. #1446 made sure that all edges that are added to the graph have a non-0 capacity. We use `maxHtlcAmount` only on edges that we fetched from the graph (results of the yen-K call).